python/glow/wgr/linear_model/ridge_model.py

```diff
@@ -312,6 +312,55 @@ def transform(self,
 
         return pivoted_df
 
+    def transform_loco(self,
+                       blockdf: DataFrame,
+                       labeldf: pd.DataFrame,
+                       sample_blocks: Dict[str, List[str]],
+                       modeldf: DataFrame,
+                       cvdf: DataFrame,
+                       covdf: pd.DataFrame = pd.DataFrame({}),
+                       chromosomes: List[str] = []) -> pd.DataFrame:
+        """
+        Generates predictions for the target labels in the provided label DataFrame by applying the model resulting from
+        the RidgeRegression fit method to the starting block matrix using a leave-one-chromosome-out (LOCO) approach.
+
+        Args:
+            blockdf : Spark DataFrame representing the beginning block matrix X
+            labeldf : Pandas DataFrame containing the target labels used in fitting the ridge models
+            sample_blocks : Dict containing a mapping of sample_block ID to a list of corresponding sample IDs
+            modeldf : Spark DataFrame produced by the RidgeRegression fit method, representing the reducer model
+            cvdf : Spark DataFrame produced by the RidgeRegression fit method, containing the results of the cross
+            validation routine.
+            covdf : Pandas DataFrame containing covariates to be included in every model in the stacking
+            ensemble (optional).
+            chromosomes : List of chromosomes for which to generate a prediction (optional). If not provided, the
+            chromosomes will be inferred from the block matrix.
+
+        Returns:
+            Pandas DataFrame containing prediction y_hat values per chromosome. The rows are indexed by sample ID and
+            chromosome; the columns are indexed by label. The column types are float64. The DataFrame is sorted using
+            chromosome as the primary sort key, and sample ID as the secondary sort key.
+        """
+        # Regex captures the chromosome name in the header
+        # level 1 header: chr_3_block_8_alpha_0_label_sim100
+        # level 2 header: chr_3_block_all_alpha_0_label_sim100
+        loco_chromosomes = chromosomes if chromosomes else [
+            r.chromosome for r in blockdf.select(
+                f.regexp_extract('header', r"^chr_([a-zA-Z0-9]+)_block", 1).alias(
+                    'chromosome')).distinct().collect()
+        ]
+        loco_chromosomes.sort()
+        print(f'Transforming with a LOCO approach against {loco_chromosomes}')
```

Comment:
Do we print a message for the other functions? It's a bit weird to do it only here. I do think we should add more log messages, but let's handle that separately.

Reply:
I've been adding logging to functions that infer missing parameters (eg. no `alpha`s) - I'll move this into the inference-specific section.